Add render pipeline

falkben opened the pull request:
basic functions to get data from render

Cleaned up the renderMeta class a little and some benchmarking

added tests and multithreading and benchmarks for multithreading

buggy integration of render stuff

img_size cannot be created on init if extents are not passed in

updating to fix conflict between master and render branches

tests passing on merge with master

Testing library switched to pytest

added get_tile test to render resource

cleaned render_resource

Add multithreading

Add time to read img stack

WIP: refactor to create IngestJob

all tests passing with new refactor

added render to IngestJob init statement

added command line args for render to ingest_large_vol and gen_commands

removed img_size as an argument, always calculated from xyz extents now

moved render resource baseURL inside the class

pass the args as a namespace variable into ingestJob and convert to dict inside ingestJob

tilewidth used as step size for getting images from render

adding tests for ingest job and render

starting to incorporate render functions into ingestJob

render stuff added to ingestJob (load img, etc.)

bug fixes

extent was being kept in the args namespace for render we want to pull it in from render no matter what the user specifies

data which was annotation data but not uint64 needs to be cast into uint64 before being posted to the BOSS

added test for ingest large vol and render

removed multithreading in ingest because intern does multithreading and it makes the code easier to maintain without it

added render scale and render window

render window isn't fully tested

render_window fine tuning

import stuff for maintining compatibility with both command line usage and pytest

negative extents are allowed with optional parameter

offsets (in order to make extents >= 0) are stored as metadata in the BOSS

Moved resources into src folder (tidying up)

Fixed a broken test

refactor of BossResParams and IngestJob

boss resources no longer stores any info about the ingest job

falkben committed:
Add render pipeline, closes #13

basic functions to get data from render

Cleaned up the renderMeta class a little and some benchmarking

added tests and multithreading and benchmarks for multithreading

buggy integration of render stuff

img_size cannot be created on init if extents are not passed in

updating to fix conflict between master and render branches

tests passing on merge with master

Testing library switched to pytest

added get_tile test to render resource

cleaned render_resource

Add multithreading

Add time to read img stack

WIP: refactor to create IngestJob

all tests passing with new refactor

added render to IngestJob init statement

added command line args for render to ingest_large_vol and gen_commands

removed img_size as an argument, always calculated from xyz extents now

moved render resource baseURL inside the class

pass the args as a namespace variable into ingestJob and convert to dict inside ingestJob

tilewidth used as step size for getting images from render

adding tests for ingest job and render

starting to incorporate render functions into ingestJob

render stuff added to ingestJob (load img, etc.)

bug fixes

extent was being kept in the args namespace for render we want to pull it in from render no matter what the user specifies

data which was annotation data but not uint64 needs to be cast into uint64 before being posted to the BOSS

added test for ingest large vol and render

removed multithreading in ingest because intern does multithreading and it makes the code easier to maintain without it

added render scale and render window

render window isn't fully tested

render_window fine tuning

import stuff for maintining compatibility with both command line usage and pytest

negative extents are allowed with optional parameter

offsets (in order to make extents >= 0) are stored as metadata in the BOSS

Moved resources into src folder (tidying up)

Fixed a broken test

refactor of BossResParams and IngestJob

boss resources no longer stores any info about the ingest job

falkben committed:
fix bug with neg z offset

moved test under src/ingest folder

perlman reviewed (approved):
Style looks good and I can easily read the code.

Only nitpick I have is the multi-level imports, e.g., "from ....ingest_large_vol import"...  It may be time for the libraryesque functions to actually be in a library.

falkben committed:
fixes bug in get img_info for negative x/y extents

formatting of gen_commands.py

better support of render resource in gen commands.py

python 3.5 support (replace exception modulenotfounderror with more general importerror)

update requirements.txt

force stride width/height to 2048 (getting from render resource, sometimes got values too large to use)

added test to check for img_info w/ neg. xy extents

falkben committed:
add multiple retries to get_tile for render resource
